sunpy/visualization/imageanimator.py

```diff
@@ -954,11 +957,25 @@ def __init__(self, data, wcs=None, image_axes=[-1, -2], unit_x_axis=None, unit_y
         list_slices_wcsaxes[image_axes[0]] = 'x'
         list_slices_wcsaxes[image_axes[1]] = 'y'
         self.slices_wcsaxes = list_slices_wcsaxes[::-1]
+
+        if(unit_x_axis is not None):
+            self.unit_x_axis = u.Unit(unit_x_axis)
+        else:
+            self.unit_x_axis = unit_x_axis
+            
+        if(unit_y_axis is not None):
+            self.unit_y_axis = u.Unit(unit_y_axis)
+        else:
+            self.unit_y_axis = unit_y_axis            
+       
+        super(ImageAnimatorWCS, self).__init__(data, image_axes=image_axes, axis_ranges=axis_ranges, **kwargs)
+
         self.unit_x_axis = unit_x_axis
         self.unit_y_axis = unit_y_axis
         super(ImageAnimatorWCS, self).__init__(data, image_axes=image_axes,
                                                axis_ranges=axis_ranges, **kwargs)
```

Comment:
I assume the above three lines should be deleted in light of the code block above?